src/Symfony/Component/Runtime/BaseRuntime.php

```diff
@@ -0,0 +1,179 @@
+        }
+
+        if (0 <= ini_get('zend.assertions')) {
+            ini_set('zend.assertions', 1);
+            ini_set('assert.active', $this->debug);
+            ini_set('assert.bail', 0);
+            ini_set('assert.warning', 0);
+            ini_set('assert.exception', 1);
+        }
+
+        set_error_handler(static function ($type, $msg, $file, $line) {
+            if ((E_DEPRECATED | E_USER_DEPRECATED) & $type) {
+                return true;
+            }
+
+            if ((error_reporting() | E_ERROR | E_RECOVERABLE_ERROR | E_PARSE | E_CORE_ERROR | E_COMPILE_ERROR | E_USER_ERROR) & $type) {
+                throw new \ErrorException($msg, 0, $type, $file, $line);
+            }
+
+            return false;
+        });
+
+        set_exception_handler(static function ($e) {
+            echo "<pre>\n$e\n</pre>\n";
+        });
+    }
+
+    public function resolve(\Closure $app): \Closure
+    {
+        $arguments = [];
+        $function = new \ReflectionFunction($app);
+
+        try {
+            foreach ($function->getParameters() as $parameter) {
+                $arguments[] = $this->getArgument($parameter, $parameter->getType());
+            }
+        } catch (\InvalidArgumentException $e) {
+            if (!$parameter->isOptional()) {
+                throw $e;
+            }
+        }
+
+        $returnType = $function->getReturnType();
+        $returnType = $returnType instanceof \ReflectionNamedType ? $returnType->getName() : '';
+
+        if ('string' === $returnType) {
+            return static function () use ($app, $arguments) {
+                return static function () use ($app, $arguments): int {
+                    echo $app(...$arguments);
+
+                    return 0;
+                };
+            };
+        }
+
+        if ('int' === $returnType || 'void' === $returnType) {
+            return static function () use ($app, $arguments) {
+                return static function () use ($app, $arguments): int {
+                    return $app(...$arguments) ?? 0;
+                };
+            };
+        }
+
+        return static function () use ($app, $arguments) {
+            if (\is_object($app = $app(...$arguments))) {
+                return $app;
+            }
+
+            $type = \function_exists('get_debug_type') ? get_debug_type($app) : \gettype($app);
+
+            if (null === $app || \is_string($app) || \is_int($app)) {
+                throw new \TypeError(sprintf('The app returned a value of type "%s" but no explicit return-type was found, did you forget to declare one?', $type));
+            }
+
+            throw new \TypeError(sprintf('The app returned a value of type "%s" while an object was expected.', $type));
+        };
+    }
+
+    public function start(object $app): \Closure
+    {
+        if ($app instanceof \Closure) {
+            if ($this->debug && (new \ReflectionFunction($app))->getNumberOfRequiredParameters()) {
```

Comment:
I'm not sure the check on debug is useful here. I would always check for the number of required param

Reply:
When the code is correct (aka always except exceptionally during dev), forcing this check all the time is costly, compared to doing nothing. That's why it's disabled n non-debug mode.